packages/backend/src/plugins/app.ts

```diff
@@ -1,14 +1,65 @@
 import { createRouter } from '@backstage/plugin-app-backend';
-import type { Router } from 'express';
+import { Router } from 'express';
 import type { PluginEnvironment } from '../types';
+import { resolvePackagePath } from '@backstage/backend-common';
+import { resolve as resolvePath } from 'path';
+import fs from 'fs-extra';
+import rateLimit from 'express-rate-limit';
 
 export default async function createPlugin(
   env: PluginEnvironment,
 ): Promise<Router> {
-  return await createRouter({
+  const appPackageName = 'app';
+
+  const appDistDir = resolvePackagePath(appPackageName, 'dist');
+  const staticDir = resolvePath(appDistDir, 'static');
+
+  const files = await fs.readdir(staticDir);
+  const jsFiles = files.filter(file => file.endsWith('.js'));
+  let injectedJSFile: string | undefined = undefined;
+
+  for (const jsFile of jsFiles) {
+    const path = resolvePath(staticDir, jsFile);
+
+    const content = await fs.readFile(path, 'utf8');
+    if (content.includes('__APP_INJECTED_')) {
+      injectedJSFile = jsFile;
+      break;
+    }
+  }
+
+  const router = await createRouter({
     logger: env.logger,
     config: env.config,
     database: env.database,
-    appPackageName: 'app',
+    appPackageName,
   });
+
+  const enclosingRouter = Router();
+  if (injectedJSFile) {
+    env.logger.info(
+      `Setting up static router for injected Javascript file ${injectedJSFile}`,
+    );
+
+    enclosingRouter.get(
+      `/static/${injectedJSFile}`,
```

Comment:
We can iterate over the array to set up multiple routes with this rule

Reply:
Is it really useful based on my answer [above](https://github.com/janus-idp/backstage-showcase/pull/710#discussion_r1383255152) ?

Reply:
Nope